packages/ui-components/src/components/tab-navigation/tab-navigation.tsx

```diff
@@ -43,57 +44,58 @@ export class KvTabNavigation {
 
 	@Element() el: HTMLKvTabNavigationElement;
 
-	initialRenderIntervalId: number;
+	observer: IntersectionObserver;
 
 	componentDidLoad() {
-		const selectedTabEl = findTabElement(this.el, this.tabs, this.selectedTabKey);
-		this.setIntervalUntilElementIsVisible(selectedTabEl);
+		this.installElementObserver(this.tabEls[this.selectedTabKey]);
 	}
 
 	disconnectedCallback() {
-		window.clearInterval(this.initialRenderIntervalId);
+		this.observer.disconnect();
 	}
 
-	private setIntervalUntilElementIsVisible = (element: HTMLStencilElement, intervalMs: number = 100): void => {
-		this.initialRenderIntervalId = window.setInterval(() => {
-			const rect = getRelativeClientRect(element);
-
-			if (rect && rect.width > 0) {
-				window.clearInterval(this.initialRenderIntervalId);
-				this.updateTabIndicatorConfig(rect);
-				return;
-			}
-		}, intervalMs);
+	private registerTabElement = (tabKey: number | string, el: HTMLKvTabItemElement) => {
+		if (this.tabEls[tabKey]) return;
+		this.tabEls[tabKey] = el;
 	};
 
-	private updateTabIndicatorConfig = (rect: DOMRect) => {
-		this.selectedTabIndicatorConfig = {
-			left: `${rect.left}px`,
-			width: `${rect.width}px`
+	private installElementObserver = (element: HTMLKvTabItemElement) => {
+		let options: IntersectionObserverInit = {
+			root: null,
+			rootMargin: '0px',
+			threshold: 1.0
 		};
+
+		this.observer = new IntersectionObserver(this.intersectionHandler, options);
+		this.observer.observe(element);
 	};
 
-	private calculateTabIndicatorPosition() {
-		const selectedTabEl = findTabElement(this.el, this.tabs, this.selectedTabKey);
+	private intersectionHandler = (entries: IntersectionObserverEntry[], observer: IntersectionObserver) => {
+		entries.forEach(intersection => {
+			const rect = getIntersectionRelativeClientRect(intersection);
 
-		if (selectedTabEl) {
-			this.updateTabIndicatorConfig(getRelativeClientRect(selectedTabEl));
-		}
-	}
+			this.selectedTabIndicatorConfig = {
+				left: `${rect.left}px`,
+				width: `${rect.width}px`
+			};
+		});
+		observer.disconnect();
```

Comment:
Seems weird disconnecting the observer here, doesn't it?
Is this because the "bug" only happens on the first load?

Reply:
this is also fired when there's a change in `selectedTabKey` so we can be certain that the element is already visible and a reliable left offset / width calculation can be made.